### ISSUE TITLE
Python error in first run (bug in Map.setCenter)

### ISSUE THREAD

**Issue by PedroVenancio** [gee-community/qgis-earthengine-plugin#33]
Dear @gena,

Thank you very much for this really great and usefull QGIS plugin!

I'm testing it in QGIS 3.10.0 on Windows 10, and in the first run of the code I always get an error.

For instance, using the landsat_median.py example:
```
import ee
from ee_plugin import Map
collection = ee.ImageCollection('LANDSAT/LC08/C01/T1_TOA')\
    .filter(ee.Filter.eq('WRS_PATH', 44))\
    .filter(ee.Filter.eq('WRS_ROW', 34))\
    .filterDate('2014-01-01', '2015-01-01')
median = collection.median()
Map.setCenter(-122.3578, 37.7726, 12)
Map.addLayer(median, {"bands": ['B4', 'B3', 'B2'], "max": 0.3}, 'median')
```

I get:
```
--- Logging error ---
Traceback (most recent call last):
  File "C:\OSGEO4~1\apps\Python37\lib\logging\__init__.py", line 985, in emit
    stream.write(msg)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:/OSGEO4~1/apps/qgis-rel-dev/./python\console\console_sci.py", line 435, in keyPressEvent
    self.entered()
  File "C:/OSGEO4~1/apps/qgis-rel-dev/./python\console\console_sci.py", line 603, in entered
    self.runCommand(self.currentCommand())
  File "C:/OSGEO4~1/apps/qgis-rel-dev/./python\console\console_sci.py", line 631, in runCommand
    more = self.runsource(src)
  File "C:/OSGEO4~1/apps/qgis-rel-dev/./python\console\console_sci.py", line 661, in runsource
    return super(ShellScintilla, self).runsource(source, filename, symbol)
  File "C:\OSGEO4~1\apps\Python37\lib\code.py", line 74, in runsource
    self.runcode(code)
  File "C:\OSGEO4~1\apps\Python37\lib\code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\Map.py", line 52, in addLayer
    ee_plugin.utils.add_or_update_ee_image_layer(image, name, shown, opacity)
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\utils.py", line 77, in add_or_update_ee_image_layer
    add_ee_image_layer(image, name, shown, opacity)
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\utils.py", line 30, in add_ee_image_layer
    check_version()
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\utils.py", line 92, in check_version
    qgis.utils.plugins['ee_plugin'].check_version()
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\ee_plugin.py", line 104, in check_version
    latest_version = requests.get('https://qgis-ee-plugin.appspot.com/get_latest_version').text
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\adapters.py", line 445, in send
    timeout=timeout
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\urllib3\connectionpool.py", line 588, in urlopen
    conn = self._get_conn(timeout=pool_timeout)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\urllib3\connectionpool.py", line 248, in _get_conn
    return conn or self._new_conn()
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\urllib3\connectionpool.py", line 823, in _new_conn
    self.num_connections, self.host, self.port or "443")
Message: 'Starting new HTTPS connection (%d): %s:%s'
Arguments: (1, 'qgis-ee-plugin.appspot.com', 443)
--- Logging error ---
Traceback (most recent call last):
  File "C:\OSGEO4~1\apps\Python37\lib\logging\__init__.py", line 985, in emit
    stream.write(msg)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:/OSGEO4~1/apps/qgis-rel-dev/./python\console\console_sci.py", line 435, in keyPressEvent
    self.entered()
  File "C:/OSGEO4~1/apps/qgis-rel-dev/./python\console\console_sci.py", line 603, in entered
    self.runCommand(self.currentCommand())
  File "C:/OSGEO4~1/apps/qgis-rel-dev/./python\console\console_sci.py", line 631, in runCommand
    more = self.runsource(src)
  File "C:/OSGEO4~1/apps/qgis-rel-dev/./python\console\console_sci.py", line 661, in runsource
    return super(ShellScintilla, self).runsource(source, filename, symbol)
  File "C:\OSGEO4~1\apps\Python37\lib\code.py", line 74, in runsource
    self.runcode(code)
  File "C:\OSGEO4~1\apps\Python37\lib\code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\Map.py", line 52, in addLayer
    ee_plugin.utils.add_or_update_ee_image_layer(image, name, shown, opacity)
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\utils.py", line 77, in add_or_update_ee_image_layer
    add_ee_image_layer(image, name, shown, opacity)
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\utils.py", line 30, in add_ee_image_layer
    check_version()
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\utils.py", line 92, in check_version
    qgis.utils.plugins['ee_plugin'].check_version()
  File "C:/Users/PedroVenancio/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\ee_plugin\ee_plugin.py", line 104, in check_version
    latest_version = requests.get('https://qgis-ee-plugin.appspot.com/get_latest_version').text
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\requests\adapters.py", line 445, in send
    timeout=timeout
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "C:\OSGEO4~1\apps\Python37\lib\site-packages\urllib3\connectionpool.py", line 393, in _make_request
    httplib_response.length)
Message: '%s://%s:%s "%s %s %s" %s %s'
Arguments: ('https', 'qgis-ee-plugin.appspot.com', 443, 'GET', '/get_latest_version', 'HTTP/1.1', 200, 5)
```
and the layer is not loaded.

After this error show up in Python console, if I just center the map again with

`Map.setCenter(-122.3578, 37.7726, 12)`

everything goes ok
![imagem](https://user-images.githubusercontent.com/1331581/70004941-e7028580-155f-11ea-9c84-996d199dbbaa.png)

After this, if I test another images, everything run smoothly, without any error.
![imagem](https://user-images.githubusercontent.com/1331581/70005138-698b4500-1560-11ea-83ca-c94643fae011.png)

Any hint about this issue?

Thank you very much!

Best regards,
Pedro

**Comment by XavierCLL** [comment]
Hi @PedroVenancio, Are you sure that you are using Qgis v2.10? the log shows that you are using Qgis v3. Because the plugin works only in Qgis v3 due that we implemented the external libraries (inside the plugin) in python 3

**Comment by XavierCLL** [comment]
I think that this is an EE authenticate "cache" error please close all Qgis instances and try again running  the same test

**Comment by PedroVenancio** [comment]
> Hi @PedroVenancio, Are you sure that you are using Qgis v2.10? the log shows that you are using Qgis v3. Because the plugin works only in Qgis v3 due that we implemented the external libraries (inside the plugin) in python 3

Hi @XavierCLL,
It was a typo, sorry, I'm using QGIS 3.10.0.

**Comment by XavierCLL** [comment]
Yes, I thought so, please check if this error is reproducible after restart Qgis

**Comment by PedroVenancio** [comment]
Hi @XavierCLL ,

Yes, this occurs after restart QGIS and even after cleaning the cache (Settings -> Options -> Network -> Cache settings). By the way, this only happens after QGIS restart, in the first Map.addLayer I do. After this first request, everything works as expected.

Checking the network request logs I get a time out:
`2019-12-03T09:32:44     WARNING    Network request https://earthengine.googleapis.com/map/aa006c10c41086d8d28bf2f43cea188d/1/0/0?token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx timed out`

I also tested in a clean profile, just with GEE plugin installed, and here the error does not show up in python console, but I also get the network request time out the first time I do Map.addLayer.

So, about the error I placed before, it can be some local conflict with other plugins I've installed. The main problem (not really a severe one) is with the request time out.

I don't know if it matters, but I'm using QGIS 3.10.0 compiled and running against GDAL/OGR 3.1.0dev and PROJ 7.0.0 (installed from OSGeo4W 64bits).

Thank you very much!

**Comment by gena** [comment]
Might be some network issue. Does the usual earthengine requests via Python API work? E.g. try running a simple Python script from a command-line using ```https://github.com/google/earthengine-api/tree/master/python``` and see if that works.

**Comment by gena** [comment]
Also, please type in the browser http://qgis-ee-plugin.appspot.com/get_latest_version, it should return 0.0.1 (current stable release version), the plugin uses this to checks for an update on the first run. If there is a timeout - probably some network/firewall issue.

**Comment by PedroVenancio** [comment]
> Checking the network request logs I get a time out:
> `2019-12-03T09:32:44 WARNING Network request https://earthengine.googleapis.com/map/aa006c10c41086d8d28bf2f43cea188d/1/0/0?token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx timed out`
> 
> I also tested in a clean profile, just with GEE plugin installed, and here the error does not show up in python console, but I also get the network request time out the first time I do Map.addLayer.

I think I've found something that can be leading to the request time out.

QGIS has EPSG:4326 as defaut coordinate reference system, and 1:37317863 as default scale.

So, going thru the script
```
import ee
from ee_plugin import Map

image = ee.ImageCollection('COPERNICUS/S2') \
  .filterDate('2017-01-01', '2017-01-02').median() \
  .divide(10000).visualize(**{'bands': ['B12', 'B8', 'B4'], 'min': 0.05, 'max': 0.5})
```
everything works as expected until this point.

`Map.setCenter(35.2, 31, 13)`
changes the default scale to 1:144448.

`Map.addLayer(image, {}, 'Sentinel-2 images January, 2018')`
changes again the scale to 1:397594653, and the EPSG to 3857, which leads to the request time out.

Doing `Map.setCenter(35.2, 31, 13)` again, zooms to the desired place and everything works again.

This behaviour is the same, if the initial EPSG is 3857, so nothing to do with the default EPSG.

So, it seems that _Map.addLayer_ is changing the zoom selected by _Map.setCenter_.

Here is a screencast where this is visible:

![eee_plugin_timeout](https://user-images.githubusercontent.com/1331581/70043406-4ee4ba80-15b8-11ea-9e16-f1c88ca0c1dc.gif)

After this, if I add another image collection -> then _Map.setCenter_ -> _Map.addLayer_, the zoom/scale is not changed by _Map.addLayer_, only the first time in the session.

Can you confirm this behaviour?

**Comment by gena** [comment]
I've noticed that during the first run Map.setCenter does not work correctly - will be addressed, thanks for reporting!!

**Comment by PedroVenancio** [comment]
> Also, please type in the browser http://qgis-ee-plugin.appspot.com/get_latest_version, it should return 0.0.1 (current stable release version), the plugin uses this to checks for an update on the first run. If there is a timeout - probably some network/firewall issue.

Hi @gena 

No, everything fine in that field
![imagem](https://user-images.githubusercontent.com/1331581/70044269-f6162180-15b9-11ea-99eb-41f3ece57ef8.png)

**Comment by PedroVenancio** [comment]
> 
> 
> I've noticed that during the first run Map.setCenter does not work correctly - will be addressed, thanks for reporting!!

Great, I'm glad that can help in some way!

I think that is _Map.addLayer_ that is changing the _Map.setCenter_ in the first run.

Thank you very much!!

**Comment by gena** [comment]
The error should be gone, fixed in #82
